scm/doc/TechGuide/chap_ccpp.rst

```diff
@@ -0,0 +1,383 @@
+(since the host model allocated its memory at initialization), but it
+will not be touching it.
+
+Adding a prognostic SCM variable
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following instructions are valid for adding a passive, prognostic
+tracer to the SCM. Throughout these instructions, the new tracer is
+called ‘smoke’.
+
+#. Add a new tracer to the SCM state. In ``ccpp-scm/scm/src/scm_type_defs.f90`` do the following:
+
+   -  Add an index for the new tracer in the ``scm_state_type`` definition.
+
+   -  Do the following in the ``scm_state_create`` subroutine:
+
+      -  Increment ``scm_state%n_tracers``
+
+      -  Set ``scm_state%smoke_index = (next available integer)``
+
+      -  Set ``scm_state%tracer_names(scm_state%smoke_index) = ‘smoke’``
+
+      -  Note: ``scm_state%state_tracer`` is initialized to zero in this subroutine already, so
+         there is no need to do so again.
+
+#. Initialize the new tracer to something other than zero (from an input
+   file).
+
+   -  Edit an existing input file (in ``ccpp-scm/scm/data/processed_case_input``): add a field in the ‘initial’
+      group of the NetCDF file(s) (with vertical dimension in pressure
+      coordinates) with an appropriate name in one (or all) of the input
+      NetCDF files and populate with whatever values are necessary to
+      initialize the new tracer.
+
+   -  Create a new input variable to read in the initialized values. In ``ccpp-scm/scm/src/scm_type_defs.f90``
+      :
+
+      -  Add a new input variable in ``scm_input_type``
+
+         .. code:: fortran
+
+            real(kind=dp), allocatable              :: input_smoke(:)
+
+      -  In ``scm_input_create``, allocate and initialize the new variable to 0.
+
+   -  Read in the input values to initialize the new tracer. In ``ccpp-scm/scm/src/scm_input.f90/get_case_init``:
+
+      -  Add a variable under the initial profile section:
+
+         .. code:: fortran
+
+            real(kind=dp), allocatable  :: input_smoke(:) !< smoke profile (fraction)
+
+      -  Add the new input variable to the allocate statement.
+
+      -  Read the values in from the file:
+
+         .. code:: fortran
+
+            call check(NF90_INQ_VARID(grp_ncid,"smoke",varID))
+                    call check(NF90_GET_VAR(grp_ncid,varID,input_smoke))
+
+      -  set ``scm_input%input_smoke = input_smoke``
+
+   -  Interpolate the input values to the model grid. Edit ``scm_setup.f90/set_state``:
+
+      -  Add a loop over the columns to call ``interpolate_to_grid_centers`` that puts ``input_smoke`` on grid levels in ``scm_state%state_tracer``
+
+         .. code:: fortran
+
+            do i=1, scm_state%n_cols
+                            call interpolate_to_grid_centers(scm_input%input_nlev, scm_input%input_pres, scm_input%input_smoke, scm_state%pres_l(i,1,:), &
+                                    scm_state%n_levels, scm_state%state_tracer(i,1,:,scm_state%smoke_index,1), last_index_init, 1)
+                    end do
```

Comment:
Indentation on this block is off

Reply:
Indentation here still off

Reply:
Sorry about that, somehow I missed these requested indentation changes. Fixed now!